dune_client/api/base.py

```diff
@@ -84,6 +91,18 @@ def _handle_response(self, response: Response) -> Any:
     def _route_url(self, route: str) -> str:
         return f"{self.base_url}{self.api_version}{route}"
 
+    def _handle_ratelimit(self, call: Callable[..., Any]) -> Any:
+        """Generic wrapper around request callables. If the request fails due to rate limiting,
+        it will retry it up to five times, sleeping i * 5s in between"""
+        for i in range(5):
+            try:
+                return call()
+            except RateLimitedError:
```

Comment:
we can use direct support on requests library for retries:
https://docs.python-requests.org/en/latest/user/advanced/#example-automatic-retries
for example:
```
from requests.adapters import HTTPAdapter
from requests.packages.urllib3.util.retry import Retry

retry_strategy = Retry(
    total=3,
    status_forcelist=[429, 500, 502, 503, 504],
    method_whitelist=["HEAD", "GET", "OPTIONS"]
)
adapter = HTTPAdapter(max_retries=retry_strategy)
http = requests.Session()
http.mount("https://", adapter)
http.mount("http://", adapter)

response = http.get("https://en.wikipedia.org/w/api.php")
```